content/flux/v0.x/data-types/basic/time.md

```diff
@@ -2,10 +2,232 @@
+
+**To convert any column to time**:
+
+1. Use [`map()`](/flux/v0.x/stdlib/universe/map/) to iterate over and rewrite rows.
+2. Use [`time()`](/flux/v0.x/stdlib/universe/time/) to convert columns values to time.
+
+```js
+data
+  |> map(fn: (r) => ({ r with epoch_ns: time(v: r.epoch_ns) }))
+```
+
+{{< flex >}}
+{{% flex-content %}}
+##### Given the following input data:
+| \_time                                            | epoch_ns _<span style="opacity:.5">(int)</span>_ |
+| :------------------------------------------------ | -----------------------------------------------: |
+| {{% nowrap %}}2021-01-01T00:00:00Z{{% /nowrap %}} |                                      10000000000 |
+| {{% nowrap %}}2021-01-01T02:00:00Z{{% /nowrap %}} |                                      20000000000 |
+| {{% nowrap %}}2021-01-01T03:00:00Z{{% /nowrap %}} |                                      30000000000 |
+| {{% nowrap %}}2021-01-01T04:00:00Z{{% /nowrap %}} |                                      40000000000 |
+{{% /flex-content %}}
+
+{{% flex-content %}}
+##### The example above returns:
+| \_time                                            | epoch_ns _<span style="opacity:.5">(time)</span>_ |
+| :------------------------------------------------ | ------------------------------------------------: |
+| {{% nowrap %}}2021-01-01T00:00:00Z{{% /nowrap %}} | {{% nowrap %}}1970-01-01T00:00:10Z{{% /nowrap %}} |
+| {{% nowrap %}}2021-01-01T02:00:00Z{{% /nowrap %}} | {{% nowrap %}}1970-01-01T00:00:20Z{{% /nowrap %}} |
+| {{% nowrap %}}2021-01-01T03:00:00Z{{% /nowrap %}} | {{% nowrap %}}1970-01-01T00:00:30Z{{% /nowrap %}} |
+| {{% nowrap %}}2021-01-01T04:00:00Z{{% /nowrap %}} | {{% nowrap %}}1970-01-01T00:00:40Z{{% /nowrap %}} |
+{{% /flex-content %}}
+{{< /flex >}}
+
+## Operate on time
+
+- [Truncate timestamps to a specified unit](#truncate-timestamps-to-a-specified-unit)
+- [Parse units of time from a timestamp](#parse-units-of-time-from-a-timestamp)
+- [Add a duration to a time value](#add-a-duration-to-a-time-value)
+- [Subtract a duration from a time value](#subtract-a-duration-from-a-time-value)
+
+### Truncate timestamps to a specified unit
+Truncating time values can be helpful when normalizing irregular timestamps.
+To truncate timestamps to a specified unit:
+
+1. Import the [`date` package](/flux/v0.x/stdlib/date/).
+2. Use [`date.truncate()`](/flux/v0.x/stdlib/date/truncate/) and provide the
+   unit of time to truncate to.
+
+{{% note %}}
+#### Truncate to weeks
+When truncating a time value to the week (`1w`), weeks are determined using the 
+**Unix epoch (1970-01-01T00:00:00Z UTC)**. The Unix epoch was on a Thursday, so
+all calculated weeks begin on Thursday.
+{{% /note %}}
+
+```js
+t0 = 2021-01-08T14:54:10.023849Z
+
+date.truncate(t: t0, unit: 1ms)
+// Returns 2021-01-08T14:54:10.023000000Z
+
+date.truncate(t: t0, unit: 1m)
+// Returns 2021-01-08T14:54:00.000000000Z
+
+date.truncate(t: t0, unit: 1mo)
+// Returns 2021-01-01T00:00:00.000000000Z
```

Comment:
would this be 2021-01-00T...?

Reply:
No, because there isn't a `00` day of the month. This is the very first nanosecond of January.